assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -820,15 +820,18 @@ private Assignment mergeAssignment(final String siteId, final Element element, f
             assignmentFromXml.setId(null);
             assignmentFromXml.setContext(siteId);
 
-            if (serverConfigurationService.getBoolean(SAK_PROP_ASSIGNMENT_IMPORT_SUBMISSIONS, false) && !assignmentFromXml.getIsGroup()) {
-                // here it's imported exactly as it was including all submissions
-                // except for group submissions as group ids will never be the same
+            if (serverConfigurationService.getBoolean(SAK_PROP_ASSIGNMENT_IMPORT_SUBMISSIONS, false)) {
                 Set<AssignmentSubmission> submissions = assignmentFromXml.getSubmissions();
-                List<String> submitters = submissions.stream().flatMap(s -> s.getSubmitters().stream()).map(AssignmentSubmissionSubmitter::getSubmitter).collect(Collectors.toList());
-                // only if all submitters can be found do we import submissions
-                if (submitters.containsAll(userDirectoryService.getUsers(submitters))) {
-                    submissions.forEach(s -> s.setId(null));
-                    submissions.forEach(s -> s.getSubmitters().forEach(u -> u.setId(null)));
+                if (submissions != null) {
```

Comment:
submissions should never be null here, why a null check isn't needed.

Reply:
If the submissions are coming back null as a result of deserialization then I would try adding:
`@JsonSetter(nulls = Nulls.AS_EMPTY)`

Reply:
> submissions should never be null here, why a null check isn't needed.

If you archive a site with an assignment with no submissions, mergeAssignment methods fails.

Reply:
Hi @ern, is this ok?